Update docs for new list aggregation functions (#6044)
mendix/docs: content/en/docs/refguide/modeling/application-logic/microflows-and-nanoflows/activities/list-activities/aggregate-list.md

In the example below, the **Return type** is **String** with the **Initial value** being `'Product names: '` and the **Expression** being `$currentResult + $currentObject/Name`.

{{<figure src="/attachments/refguide/modeling/application-logic/microflows-and-nanoflows/activities/list-activities/aggregate-list/aggregate-list-reduce-example.png" width="500px" alt="Aggregate list reduce example" >}}

### 3.4 Aggregate with

This option is only visible if the **Function** is **Average**, **Minimum**, **Maximum** or **Sum**. For those functions, you can define if you want to aggregate using an attribute or an expression.

{{% alert color="info" %}}
When using the `Count` function, it is not necessary to select an attribute or enter an expression, as it simply counts the number of objects in the list.
{{% /alert %}}

### 3.5 Attribute

Defines which attribute of the objects in the list is used to aggregate over. This must be a numeric attribute (**Long**, **Integer**, or **Decimal**). This option is only visible if **Aggregate with** is set to **Attribute**.

### 3.6 Initial Value

> This option is only visible if the **Function** is **Reduce**. The expression here will be used as the initial value for the aggregation. When aggregating with **Reduce** over an empty list, the result will be this initial value. Otherwise this initial value will be the value of `$currentResult` when the **Expression** is evaluated for the first item.

Reviewer: @WimJongeneel In this sentence `The expression here will be used as the initial value for the aggregation.`, do you mean that the result of the whole expression will be used as the initial value?

I am a bit confused, because from this sentence `Otherwise this initial value will be the value of $currentResult when the Expression is evaluated for the first item`, it seems that only the value of `$currentResult` will be used as the initial value. How about when the expression is `$currentResult + $currentObject/Name` in the example you gave above?

Reviewer: @WimJongeneel In this sentence` When aggregating with **Reduce** over an empty list, the result will be this initial value.`, can we be a bit more explicit about the result here? Like, what kind of result usually it is when aggregating with **Reduce** over an empty list?

Author: > @WimJongeneel In this sentence The expression here will be used as the initial value for the aggregation., do you mean that the result of the whole expression will be used as the initial value?

Yes, that is true. The entire expression in the 'Initial expression' field will be evaluated first and will be the value of `$currentResult` before we iterate over the items in the list.

After that it will iterate over the items in the list and the value of `$currentResult` will be updated to the result of the expression in the 'Expression' field (for each step in the iteration). Within this expression you can use the current value of `$currentResult` to create a new value for `currentResult`. 

So in the example and for a list `[ { Name: 'Product_1' }, { Name: 'Product_2' }, { Name: 'Product_3' }  ]` the value of `$currentResult` will go through 4 values:

* `'Product names: '`: after evaluating the initial expression and before iterating
* `'Product names: Product_1'`: previous value + `$currentObject/Name` where `$currentObject` is the first item in the list
* `'Product names: Product_1Product_2'`: previous value + `$currentObject/Name` where `$currentObject` is the second item in the list
* `'Product names: Product_1Product_2Product_3'`: previous value + `$currentObject/Name` where `$currentObject` is the third item in the list

And the last value is then the result of the aggregation.

For an empty list it is the result of evaluating is the initial expression because nothing more will happen after the first step.

Reviewer: @WimJongeneel Thanks a lot for the explanation. I think I mixed up the expression you put in the **Initial value** field and the expression you put in the **Expression** field.

 I just made the distinguishment a bit more clear in the doc. From my understanding, the initial value will first be the result of the entire expression in the **Initial value** field. If the aggregation is not over an empty list, the initial value will then be updated to the value of `$currentResult` when **Expression** is evaluated for the first item. Is my understanding correct? Please see my update in [this ](https://github.com/mendix/docs/pull/6044/commits/50a4da2a6ae7fa8e4aeabe39d68cd667570ec744)commit.

Author: Yes, looks good from my side

Reviewer: Great!

| Minimum | The minimum of all values of an attribute from the list of objects.            |
| Maximum | The maximum of all values of an attribute from the list of objects.            |
| Sum     | The sum of all values of an attribute from the list of objects.                |
| All     | If an expression is true for all objects in the list of objects                |
| Any     | If an expression is true for at least any of the object in the list of objects |
| Reduce  | Allows for a custom aggregation to be defined                                  |

3.3 Return type

This option is only visible if the Function is Reduce. The type selected here will be the type of the output variable, the type of the 'Initial expression' and the type of the 'Expression'. In the example shown below the 'Return type' is String with the 'Initial value' being `'Product names: '` and the expression being `$currentResult + $currentObject/Name`.

{{<figure src="/attachments/refguide/modeling/application-logic/microflows-and-nanoflows/activities/list-activities/aggregate-list/aggregate-list-reduce-example.png" width="500px" alt="Aggregate list reduce example" >}}

3.3 Aggregate with

> This option is only visible for if the Function is Average, Minimum, Maximum or Sum. For those function you can here define if you want to aggregate with an attribute or an expression. For All, Any and Reduce this option is hidden because they only support expressions.

Reviewer 2: ```suggestion
This option is only visible if the Function is Average, Minimum, Maximum or Sum. For those functions, you can define if you want to aggregate using an attribute or an expression.
```

You don't need to mention that it is hidden for the rest. Also fixed a typo.

{{% alert color="info" %}}
When using the `Count` function, it is not necessary to select an attribute or enter an expression, as it simply counts the number of objects in the list.
{{% /alert %}}

### 3.4 Attribute

Defines which attribute of the objects in the list is used to aggregate over. This must be a numeric attribute (Long, Integer, or Decimal). This option is only visible if 'Aggregate with' is set to 'attribute'.

### 3.4 Initial value

This option is only visible for if the 'Function' is Reduce. The expression here will be used as the initial value for the aggregation. When aggregating with Reduce over an empty list the result will be this value. Oherwise this value will be the value of `$currentResult` when the 'Expression' is evaluated for the first item.

> ### 3.4 Expression

Reviewer 2: This and the next section have incorrect section numbers.
